### ISSUE TITLE
[FEATURE] Allow users to override Packages/.gitignore

### ISSUE THREAD

**Issue by Faxmashine** [vrchat-community/creator-companion#416]
Currently, the VCC adds the following two lines to every project's `Packages/.gitignore`.

```
/*/
!com.vrchat.core.*/
```

This happens silently and disrupts the workflow of some users.

https://feedback.vrchat.com/creator-companion/p/dont-gitignore-entire-packages-folder

Users should be allowed to prevent the VCC from modifying `Packages/.gitignore`. Alternatively, the VCC could display or prompt or warning to inform users about modifications to the `.gitignore` file.

**Comment by ChildoftheBeast** [comment]
i would love to see a flag for this getting added to VCC, as this disrupts all of my collaborative projects.

**Comment by advanced4** [comment]
yeah i'd really love to see this fixed, im not really sure what the purpose of this is because it just breaks every shared project

**Comment by techanon** [comment]
For the time being, the work around is to explicitly unignore all prefix packages that you have in your project, and the re-ignore specific packages you don't want to commit.  
For example:
```
/*/
!com.vrchat.core.*/
!at.*/
!com.*/
!dev.*/
!sh.*/
!tlp.*/
dev.onevr.vrworldtoolkit/
```
VCC currently just tries to make sure that the first two lines listed are in the gitignore. So we include them and add the more specific exclusions and inclusions as needed.

**Comment by orels1** [comment]
The idea behind this was always to allow the VPM to re-resolve the pacakges instead of storing them in git directly, but due current resolve limitations I think it makes sense to make that optional for now

**Comment by ChildoftheBeast** [comment]
> The idea behind this was always to allow the VPM to re-resolve the pacakges instead of storing them in git directly, but due current resolve limitations I think it makes sense to make that optional for now

i'm still not sure why it should work like this.
I get having the VPM manger pull the packages if they're not there, but i'm not sure as to why the gitignore needs to be there, as i don't feel that the package manager should be dictating what should and shouldn't go in to a github repo. I feel like that should be left up to the end user to decide and not be enforced, or at very minimum have a global opt-out for all your projects.
